IPython/core/ultratb.py

```diff
@@ -1094,12 +1094,14 @@ def structured_traceback(self, etype, evalue, etb, tb_offset=None,
                 etype, evalue, etb = exception
             else:
                 evalue = None
+            chained_exc_ids = set()
```

Comment:
I think this misses the first exception value we prepare the traceback for. Of course, if there's a cycle, we'll still pick it up one step later, but I think it would be a bit neater to start the set with evalue in.